intro-HPC/ch_easybuild.tex

```diff
@@ -0,0 +1,150 @@
+
+\begin{itemize}
+    \item \lstinline|$VSC_DATA/easybuild/sources| is where EasyBuild will (try to) automatically
+        download sources if they're not available yet
+    \item \lstinline|/apps/gent/source| is the central ``cache'' for already downloaded sources,
+        and will be considered by EasyBuild before downloading anything
+\end{itemize}
+
+
+
+
+\subsection{Build directory}
+
+This directory is where EasyBuild will build software in. To have good performance,
+this needs to be on a fast filesystem.
+
+\begin{prompt}
+%\shellcmd{export EASYBUILD\_BUILDPATH=\${TMPDIR:-/tmp/\$USER}}%
+\end{prompt}
+
+On cluster nodes, you can use the fast, in-memory \lstinline|/dev/shm/$USER| location
+as a build directory.
+
+\subsection{Software install location}
+
+This is where EasyBuild will install the software (and accompanying modules) to.
+
+For example, to let it use \lstinline|$VSC_DATA/easybuild|, use:
+
+
+% Don't put the whole command inside \shellcmd{} because it would show ugly white boxes
+
+\begin{prompt}
+%\shellcmd{export}% EASYBUILD_INSTALLPATH=$VSC_DATA/easybuild/$VSC_OS_LOCAL/$VSC_ARCH_LOCAL$VSC_ARCH_SUFFIX
+\end{prompt}
+
+Using the \lstinline|$VSC_OS_LOCAL|, \lstinline|$VSC_ARCH| and \lstinline|$VSC_ARCH_SUFFIX| environment variables
+ensures that your install software to a location that is specific to the cluster you are
+building for.
+
+Make sure you \strong{do not build software on the login nodes}, since the loaded \lstinline|cluster|
+module determines the location of the installed software. Software built on the login
+nodes may not work on the cluster you want to use the software on (see also \autoref{sec:running-software-incompatible-with-host}).
+
+To share custom software installations with members of your VO, replace \lstinline|$VSC_DATA|
+with \lstinline|$VSC_DATA_VO| in the example above.
+
+\section{Using EasyBuild}
+
+Before using EasyBuild, you first need to load the \lstinline|EasyBuild| module. We don't specify
+a version here (this is an exception, for most other modules you should, see \autoref{subsec:explicit-version-numbers}) because newer versions
+might include important bug fixes.
+
+\begin{prompt}
+module load EasyBuild
+\end{prompt}
+
+\subsection{Installing supported software}
+
+EasyBuild provides a large collection of readily available software versions,
+combined with a particular toolchain version. Use the \lstinline|--search|
+(or \lstinline|-S|) functionality to see which different 'easyconfigs'
+(build recipes, see \url{http://easybuild.readthedocs.org/en/latest/Concepts_and_Terminology.html#easyconfig-files}) are available:
+
+\begin{prompt}
+%\shellcmd{eb -S example-1.2}%
+CFGS1=/apps/gent/CO7/sandybridge/software/EasyBuild/3.6.2/lib/python2.7/site-packages/easybuild_easyconfigs-3.6.2-py2.7.egg/easybuild/easyconfigs
+ * $CFGS1/e/example/example-1.2.1-foss-%\the\year{}%a.eb
+ * $CFGS1/e/example/example-1.2.3-foss-%\the\year{}%b.eb
+ * $CFGS1/e/example/example-1.2.4_libexample.patch
```

Comment:
kick out this line, it's actually a bug that patches show up in the output of `--search`...